app/javascript/components/student/solution-summary/Header.tsx

```diff
@@ -0,0 +1,190 @@
+  links: SolutionSummaryLinks
+}): JSX.Element => {
+  switch (iteration.status) {
+    case IterationStatus.TESTING:
+    case IterationStatus.ANALYZING:
+      return (
+        <header>
+          <div className="info">
+            <h2>Your solution is being processed...</h2>
+            <p>
+              Your solution is currently being tested, analysed, and compared to
+              other solutions to find for potential improvements. This should
+              only take a few seconds.
+            </p>
+          </div>
+          <GraphicalIcon icon="spinner" className="spinner" />
+        </header>
+      )
+    case IterationStatus.TESTS_FAILED:
+      return (
+        <header>
+          <div className="info">
+            <h2>Your solution failed the tests.</h2>
+            <p>
+              Hmmm, it looks like your solution isn&apos;t working. Please check
+              that the tests pass locally on your machine. If they do pass for
+              you, please read{' '}
+              <a
+                href={links.testsPassLocallyArticle}
+                rel="noreferrer"
+                target="_blank"
+              >
+                this article{' '}
+                <Icon icon="external-link" alt="Opens in a new tab" />
+              </a>
+            </p>
+          </div>
+          <div className="status failed">Tests Failed</div>
+        </header>
+      )
+    case IterationStatus.ESSENTIAL_AUTOMATED_FEEDBACK: {
+      const comments = [
+        `${iteration.numEssentialAutomatedComments} essential ${pluralize(
+          'improvement',
+          iteration.numEssentialAutomatedComments
+        )}`,
+        iteration.numActionableAutomatedComments > 0
+          ? `${iteration.numActionableAutomatedComments} ${pluralize(
+              'recommendation',
+              iteration.numActionableAutomatedComments
+            )}`
+          : '',
+        iteration.numNonActionableAutomatedComments > 0
+          ? `${
+              iteration.numNonActionableAutomatedComments
+            } additional ${pluralize(
+              'comment',
+              iteration.numNonActionableAutomatedComments
+            )}`
+          : '',
+      ].filter((comment) => comment.length > 0)
+
+      return (
+        <header>
+          <div className="info">
+            <h2>Your solution worked, but you can take it further...</h2>
+            <p>
+              We’ve analysed your solution and have {toSentence(comments)}.
+              Address the essential improvements before proceeding.
+            </p>
+          </div>
+          <div className="status passed">Tests Passed</div>
+        </header>
+      )
+    }
+    case IterationStatus.NO_AUTOMATED_FEEDBACK:
+      return (
+        <header>
+          <div className="info">
+            <h2>Your solution looks great!</h2>
+            <p>
+              Your solution passed the tests and we&apos;ve not got any
+              recommendations.
```

Comment:
I don't know if this makes sense, but to me the following is easier to read:

```suggestion
              Your solution passed the tests and we don't have any
              recommendations.
```

Reply:
I've happy with this suggestion. That said, all the copy will be reviewed before launch.

Reply:
I've fixed this :)